src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -186,11 +189,94 @@ public DynamoView(DynamoViewModel dynamoViewModel)
                  }
              };
 
+            // Add an event handler to check if the collection is modified.   
+            TabItems.CollectionChanged += this.OnCollectionChanged;
+
+            this.HideOrShowRightSideBar();
+
             this.dynamoViewModel.RequestPaste += OnRequestPaste;
             this.dynamoViewModel.RequestReturnFocusToView += OnRequestReturnFocusToView;
             FocusableGrid.InputBindings.Clear();
         }
 
+        // This method adds a tab item to the right side bar and 
+        // sets the extension window as the tab content.
+        internal TabItem AddTabItem(IViewExtension viewExtension, ContentControl contentControl)
+        {
+            int count = TabItems.Count;
+
+            if (!IsExtensionAddedToRightSideBar(viewExtension))
+            {
+                tabDynamic.DataContext = null;
+
+                // creates a new tab item
+                TabItem tab = new TabItem();
+                tab.Header = viewExtension.GetType().Name;
+                tab.Tag = viewExtension.GetType();
+                tab.HeaderTemplate = tabDynamic.FindResource("TabHeader") as DataTemplate;
+
+                // setting the extension window to the current tab content
+                tab.Content = contentControl.Content;
```

Comment:
@QilongTang @reddyashish - did you guys have a chance to discuss this last week?

Reply:
Yes, I am looking into this issue.

Reply:
@mjkkirschner Changing this to set the contentControl object to tab.Content, worked(only for UserControl elements). This would then fail if the user is the creating a window type extension. 

I will be making this change in a new PR, but the users will have a constraint of injecting user control elements only.